router/chi/endpoint.go

```diff
@@ -0,0 +1,148 @@
+package chi
```

Comment:
I think you can reuse a lot of logic from the `mux` package by just injecting your required param extractor.

```
// HandlerFactory creates a handler function that adapts the chi router with the injected proxy
type HandlerFactory func(*config.EndpointConfig, proxy.Proxy) http.HandlerFunc

func NewEndpoint(cfg *config.EndpointConfig, prxy proxy.Proxy) http.HandlerFunc {
    hf := mux.CustomEndpointHandler(
        mux.NewRequestBuilder(func(r *http.Request) map[string]string {
            return extractParamsFromEndpoint(r, cfg.Endpoint)
        }),
    )
    return hf(cfg, prxy)
}

func extractParamsFromEndpoint(r *http.Request, endpoint string) map[string]string {
    ctx := r.Context()
    rctx := chi.RouteContext(ctx)

    params := map[string]string{}
    if len(rctx.URLParams.Keys) > 0 {
        for _, param := range rctx.URLParams.Keys {
            params[strings.Title(param)] = chi.URLParam(r, param)
        }
    }
    return params
}
```